app/main.js

```diff
@@ -45,7 +49,7 @@ app.on('ready', startProcessWin)
 app.on('ready', loadSettings)
 app.on('ready', createTrayIcon)
 app.on('ready', startPowerMonitoring)
-
+app.on('ready', createWindow)
```

Comment:
I think that this is why I did  not see the Welcome page when I first tried it: 
- when app starts all these `app.on('ready'` are called, but as we are in nodejs, they are not called in sync way (one after another) 
- settings were not loaded yet, so `settings.get('showWelcomeWindow')` was something like `null` and I did not get window

Your solution works, but it's way too hacky: 
```
+  settings = new AppSettings(settingsFile)
+  if(settings.get('showWelcomeWindow') == null){
+    settings.set('showWelcomeWindow', true)
+  }
```

Instead we should create welcome window once settings are loaded (at the end of `function loadSettings ()`) or try to find if we can load settings before `app.on('ready', )`